src/frontend/apps/impress/src/features/docs/doc-tree/hooks/useTreeUtils.tsx

```diff
@@ -1,14 +1,10 @@
-import { useTreeContext } from '@gouvfr-lasuite/ui-kit';
-
 import { Doc } from '@/docs/doc-management';
 
 export const useTreeUtils = (doc: Doc) => {
-  const treeContext = useTreeContext<Doc>();
-
   return {
-    isParent: doc.nb_accesses_ancestors <= 1, // it is a parent
-    isChild: doc.nb_accesses_ancestors > 1, // it is a child
-    isCurrentParent: treeContext?.root?.id === doc.id || doc.depth === 1, // it can be a child but not for the current user
+    isTopParent: doc.depth === 1,
+    isChild: doc.depth > 1,
+    isCurrentTopParent: typeof doc.children === 'undefined', // it can be a child but not for the current user
```

Comment:
if a doc has no children, it can be a parent. Not sure to understand what you are looking for.

Reply:
The doc response came from the endpoint `documents/[DOC_ID]/`, it is why it was without `child` array.

There is not a reliable way to know if a doc is the top root without context, I use another way to get the information that I need so:
https://github.com/suitenumerique/docs/blob/e8147d02c1bb9eeea86882a6425ab39be7c0929e/src/frontend/apps/impress/src/features/docs/doc-tree/components/DocTree.tsx#L210